dot/rpc/modules/chain_test.go

```diff
@@ -2,3 +2,529 @@
 // SPDX-License-Identifier: LGPL-3.0-only
 
 package modules
+
+import (
+	"errors"
+	"math/big"
+	"net/http"
+	"testing"
+
+	apimocks "github.com/ChainSafe/gossamer/dot/rpc/modules/mocks"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestChainModule_GetBlock(t *testing.T) {
+	testHash := common.NewHash([]byte{0x01, 0x02})
+	inputHash := common.MustHexToHash("0x0102000000000000000000000000000000000000000000000000000000000000")
+	emptyBlock := types.NewEmptyBlock()
+
+	mockBlockAPI := new(apimocks.BlockAPI)
+	mockBlockAPI.On("GetBlockByHash", inputHash).Return(&emptyBlock, nil)
+	mockBlockAPI.On("BestBlockHash").Return(testHash, nil)
+
+	mockBlockAPIGetHashErr := new(apimocks.BlockAPI)
+	mockBlockAPIGetHashErr.On("GetBlockByHash", inputHash).Return(nil, errors.New("GetJustification error"))
+
+	bodyBlock := types.NewEmptyBlock()
+	bodyBlock.Body = types.BytesArrayToExtrinsics([][]byte{{1}})
+	mockBlockAPIWithBody := new(apimocks.BlockAPI)
+	mockBlockAPIWithBody.On("GetBlockByHash", inputHash).Return(&bodyBlock, nil)
+
+	expEmptyBlock := &ChainBlockResponse{Block: ChainBlock{
```

Comment:
nit, move these within the test structs if they aren't reused by multiple tests.